src/test/util_tests.cpp

```diff
@@ -1465,6 +1466,40 @@ BOOST_AUTO_TEST_CASE(test_IsDigit)
     BOOST_CHECK_EQUAL(IsDigit(9), false);
 }
 
+/* Check for overflow */
+template <typename T>
+static bool TestAddMatrixOverflow()
+{
+    constexpr T MAXI{std::numeric_limits<T>::max()};
+    assert(!CheckedAdd(T{1}, MAXI));
+    assert(!CheckedAdd(MAXI, MAXI));
+    assert(0 == CheckedAdd(T{0}, T{0}).value());
```

Comment:
These can be `BOOST_CHECK()` and `BOOST_CHECK_EQUAL()`, the `TestAddMatrixOverflow()` and `TestAddMatrix()` functions be `void` and can be called without `BOOST_CHECK()` from the test case.

This would have the usual benefit of using `BOOST_CHECK()` vs `assert()`.

Reply:
nice, fixed.